rsql-jpa/src/main/java/io/github/perplexhub/rsql/RSQLJPAPredicateConverter.java

```diff
@@ -265,43 +308,62 @@ public Predicate visit(ComparisonNode node, From root) {
 			Comparable comparable = (Comparable) argument;
 
 			if (op.equals(GREATER_THAN)) {
-				return builder.greaterThan(attrPath, comparable);
+				return builder.greaterThan(expr, comparable);
 			}
 			if (op.equals(GREATER_THAN_OR_EQUAL)) {
-				return builder.greaterThanOrEqualTo(attrPath, comparable);
+				return builder.greaterThanOrEqualTo(expr, comparable);
 			}
 			if (op.equals(LESS_THAN)) {
-				return builder.lessThan(attrPath, comparable);
+				return builder.lessThan(expr, comparable);
 			}
 			if (op.equals(LESS_THAN_OR_EQUAL)) {
-				return builder.lessThanOrEqualTo(attrPath, comparable);
+				return builder.lessThanOrEqualTo(expr, comparable);
 			}
 		}
 		log.error("Unknown operator: {}", op);
 		throw new RSQLException("Unknown operator: " + op);
 	}
 
-	private Predicate equalPredicate(Path attrPath, Class type, Object argument) {
+	private Expression<?> getJsonExpression(Path<?> path, Attribute attribute, ComparisonNode node) {
+		var database = getDatabase(attribute).orElse(null);
+
+		if (database == Database.POSTGRESQL) {
+			var args = new ArrayList<Expression<?>>();
+			args.add(path);
+
+			Stream.of(node.getSelector().split("\\."))
+					.skip(1) // skip root
+					.map(builder::literal)
+					.map(expr -> expr.as(String.class))
+					.forEach(args::add);
+
+			return builder.function("jsonb_extract_path_text", String.class, args.toArray(Expression[]::new));	
```

Comment:
I think this is great, but it would be awesome to have type match as well, I ended up doing the same thing before you guys decided to support it and the way read data is like `jsonb_extract_path_text('jsonbColName', path1, subpath...)::Type'

Reply:
@nstdio

Reply:
How we can determine what `SQLType` is?

Reply:
sorry for the late reply, so the way i did was determining data type from the value passed in by passing it to chain of parsers , this is assuming the consumer of rsql api knows the data type they are want to query inside jsonb, its not very effective but another way would be to make rsql accept datatype indentifier

static {
    parsers.put("LONG", Long::parseLong);
    parsers.put("FLOAT", Double::parseDouble);
    parsers.put("BOOLEAN", BooleanUtils::toBooleanObject);
    parsers.put("TIMESTAMPTZ", DateUtil::parseToTimeZoneFormat);
  }